src/realm/sync/config.cpp

```diff
@@ -54,11 +54,20 @@ bool SyncError::is_client_reset_requested() const
     if (error_code.category() != realm::sync::protocol_error_category()) {
         return false;
     }
-    // Documented here: https://realm.io/docs/realm-object-server/#client-recovery-from-a-backup
-    return (error_code == ProtocolError::bad_server_file_ident ||
-            error_code == ProtocolError::bad_client_file_ident || error_code == ProtocolError::bad_server_version ||
-            error_code == ProtocolError::diverging_histories || error_code == ProtocolError::client_file_expired ||
-            error_code == ProtocolError::invalid_schema_change);
+    // clang-format off
+    // keep this list in sync with SyncSession::handle_error()
```

Comment:
hmm. In my view, `SyncError::is_client_reset_requested()` is the helper function and it is already called from the place that needs to keep this in sync. The reason the codes are duplicated is because I would like to keep the switch statement on all possible `ProtocolError` codes in `SyncSession::handle_error()`. The nice property about that is that the compiler will warn us about it if we miss handling one there. With this perspective, do you still see a need for change?